File: Baekjoon/S5/C#/S5_셀프 넘버.cs
https://www.acmicpc.net/problem/4673

using System;
using System.Text;

namespace Baekjoon
{
    class Program
    {
        static void Main(string[] args)
        {
            StringBuilder sb = new StringBuilder();
            bool[] arr = new bool[10001];
            for (int i = 1; i <= 10000; i++)
            {
                if (cr(i) <= 10000)
                    arr[cr(i)] = true;
                if (arr[i] == false)
                    sb.Append(i + "\n");
            }
            Console.WriteLine(sb);
        }

        static int cr(int a)
        {
            int m = a;
            while (a > 0)
            {
                m = m + a % 10;
                a /= 10;
            }
            return m;
        }
    }
}

File: Baekjoon/S5/JAVA/S5_셀프 넘버.java
//https://www.acmicpc.net/problem/4673

import java.io.*;

public class Main{
    public static void main(String args[]) throws IOException{
        boolean[] ans = new boolean[10001];
        StringBuilder sb = new StringBuilder();
        
        int idx = 0;
        for(int i = 1; i <= 10000; i++){
            idx = check(i);
            
            if(idx <= 10000)
                ans[check(i)] = true;
            
            if(!ans[i])
                sb.append(i+"\n");
        }
        
        System.out.println(sb);
    }
    
    static int check(int n){
        int sum = n;
        while(n > 0){
            sum += n%10;
            n /= 10;
        }
        return sum;
    }
}